### ISSUE TITLE
ECC x509 keys

### ISSUE THREAD

**Issue by jcharaoui** [duritong/trocla#82]
It would be great to be able to create ECC x509 keys with trocla.

ECC keys are widely supported now and benefit from much smaller size compared to RSA based keys.

**Comment by duritong** [comment]
Absolutely, would be happy to merge such a MR. Important would be to make it non-breaking for existing things. But given that these use-cases are quite heavily tested, I would expect to be able to hold a stable interface.

**Comment by duritong** [comment]
A new option called type (or so) could be introduced, which defaults to RSA and then it could be switched to ECC and then based on that option we could either dispatch to `OpenSSL::PKey::RSA`or `OpenSSL::PKey::EC`.

And it would require some detection when re-constructing the key based on what we have stored.
